tools/proto2jsonschema/gen.sh

```diff
@@ -0,0 +1,99 @@
+#!/bin/bash
+
+set -e
+
+# Only run in repo root
+currentDir=${PWD##*/}
+
+if [ "$currentDir" != "google-cloudevents" ] ; then
+  echo "Error: Script must be run in google-cloudevents root.";
+  exit 1;
+fi
+
+# Install protoc
+case "$OSTYPE" in
+  linux*)
+    PROTOBUF_PLATFORM=linux-x86_64
+    PROTOC=tmp/bin/protoc
+    ;;
+  win* | msys* | cygwin*)
+    PROTOBUF_PLATFORM=win64
+    PROTOC=tmp/bin/protoc.exe
+    ;;
+  darwin*)
+    PROTOBUF_PLATFORM=osx-x86_64
+    PROTOC=tmp/bin/protoc
+    ;;
+  *)
+    echo "Unknown OSTYPE: $OSTYPE"
+    exit 1
+esac
+
+rm -rf tmp
+mkdir tmp
+cd tmp
+echo "- Downloading protobuf tools"
+PROTOBUF_VERSION=3.12.3
+curl -sSL \
+  https://github.com/protocolbuffers/protobuf/releases/download/v$PROTOBUF_VERSION/protoc-$PROTOBUF_VERSION-$PROTOBUF_PLATFORM.zip \
+  --output protobuf.zip
+unzip -q protobuf.zip
+cd ..
+
+echo "- Setting up protoc plugin (chrusty/protoc-gen-jsonschema)"
+go get -v github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema
+go install github.com/chrusty/protoc-gen-jsonschema/cmd/protoc-gen-jsonschema
+
+echo "- Converting protos to JSON Schemas"
+chmod +x $PROTOC
+
+# Generate JSON Schemas
+OUT_DIR="jsonschema"
+
+## Get all data.proto files
+DATA_PROTOS=$(find proto/google/events -name "data.proto")
+for proto in $DATA_PROTOS; do
+  ## For each data.proto file, generate the jsonschema
+  # proto e.g. proto/google/events/cloud/pubsub/v1/data.proto
+
+  # Cut proto/ prefix
+  PATH_PREFIX="proto/"
+  PATH_PREFIX_LEN=$((1 + $(echo ${#PATH_PREFIX})))
+
+  # e.g. google/events/cloud/pubsub/v1/data.proto
+  CUT_PROTO=$(echo "$proto" | cut -c $PATH_PREFIX_LEN-)
+
+  # e.g. google/events/cloud/pubsub/v1/
+  PATH_POSTFIX="/data.proto"
+  PATH_POSTFIX_LEN=$((1 + $(echo ${#PATH_POSTFIX})))
+  CUT_PROTO_DIR=$(echo "$CUT_PROTO" | rev | cut -c $PATH_POSTFIX_LEN- | rev)
```

Comment:
Another bash substitution option:

```suggestion
  PATH_POSTFIX="/data.proto"
  CUT_PROTO_DIR=${proto%"$PATH_POSTFIX"}
```